Added bazel rule with example to test graph serialization

margaritiko opened the pull request:
Add bazel rule to extract `.textproto` file from app jar

Now all process looks like this:
* In dagger spi plugin with `Filer` instance we save a file with binding graph to `binding_graph.textproto`
* This file with graph appears in main app's jar
* With rule `dagger_query_textproto` we extract file from this jar and now able to add it to any other target 🎉

margaritiko committed:
Added bazel rule with example to test graph serialization

kwchan1995 commented on project/src/com/google/daggerquery/dagger_query_textproto.bzl:
```diff
@@ -0,0 +1,28 @@
+        inputs = src_jars,
+        outputs = [ctx.outputs.out],
+        command = "unzip -p -o {src_jar} binding_graph.textproto > {out}".format(
+            src_jar = src_jars[0].path,
```
Sorry Rita I ran into problems setting this up internally so I wasn't able to debug this. I think what you can do is to print out the path of this `src_jar`, and manually examine the contents of this jar to ensure the `binding_graph.textproto` file is actually there. This will help narrow down whether it's a problem with the shell command, we're feeding in the wrong jar, or there was a problem packaging the textproto file into the jar.

margaritiko replied:
No problem, it's good idea, thanks!
I will try to reproduce these steps 👌

margaritiko commented:
@Ratchette @kwchan1995 @allie-wood  
Finally I solved this problem and now rule is correct.
I've removed all example sources and PR became smaller 😊

I also updated description of this PR with up-to-date information about process of file's generation ✨

kwchan1995 commented on project/src/com/google/daggerquery/dagger_query_textproto.bzl:
```diff
@@ -0,0 +1,42 @@
+        inputs = src_jars,
+        outputs = [ctx.outputs.out],
+        command = "unzip -p {src_jar} binding_graph.textproto > {out}".format(
+            src_jar = src_jars[0].path,
```
May be worth adding an assert statement that `src_jars` is of length 1. This way it can fail earlier for easier debugging instead of the error silently propagated downstream.

margaritiko replied:
Done ✅

kwchan1995 reviewed (approved):
Awesome!

Ratchette reviewed (approved):
LGTM 🦙

margaritiko committed:
Add failure if there are multiple jars with sources
